modules/ROOT/pages/client-id-based-policies.adoc

```diff
@@ -14,48 +14,73 @@ endif::[]
+* Generate the client ID and client secret credentials from Anypoint Platform.
++
+Using client credentials generated outside of Anypoint Platform fails the HTTP request.
+* Provide the client ID and client secret pair of the registered client application.
+* If using REST console, create a trait in the RAML root and then reference it in every operation of your API affected by policy.
++
+To obtain the RAML code to add to the policy, go to the corresponding policy in your API Manager.
+//It looks like we might need this for other policies too? If so, we should move it out of here and put it in a central place.
++
+image::raml-snippet.png[]
 
-= REST Console
-To prevent user requests from being rejected when using the REST Console, create a trait in the RAML root and then reference it in every operation of your API affected by policy. You may find the RAML snippet link containing the RAML code you need to add to the RAML in the corresponding policy from the list of applied policies in the Policies tab of your API specification in API Manager.
 
-image::raml-snippet.png[]
+== Policy Implementation Best Practices
+
+If you need to track client application access to your API, apply the Client ID Enforcement policy to makes sure that the client credentials are provided in all HTTP requests. 
+
+//This may be too little info to warrant a section of its own. Let's reassess.
 
-= Credentials source
 
-The policy can be configured to extract the client ID and client secret from the HTTP request in different ways using DataWeave 2.0 expressions.
+== Configure the Policy
 
-The following examples assume that the values for application credentials are '1234' for client ID and 'abcd' for client secret.
+You can configure the policy to extract the client ID and client secret from the HTTP request in different ways using DataWeave 2.0 expressions. If when configuring the policy, the client secret DataWeave 2.0 expression is not specified, then only the client ID specified in the HTTP request is validated against the protected resource.
 
-NOTE: If when configuring the policy, client secret DataWeave 2.0 expression is specifically not specified (ie empty field), then only the client ID specified on the HTTP requests against the protected resource will be validated.
+When an HTTP request is performed against a protected resource for which the Client ID Enforcement policy is applied, and the client application credentials are invalid or unauthorized, the HTTP response includes a WWW Authenticate header with the following values:
+
+[width="100%", cols="2,70"]
+|===
+>s| Custom mode | Header 'WWW-Authenticate'='Client-ID-Enforcement'
+>s| Basic Auth mode | Header 'WWW-Authenticate'='Basic realm="mule-realm"
+|===
 
-== Credentials as HTTP headers
+
+=== Configure Parameters
```

Comment:
I would remove this and have the sections as part of the examples of the configuration of the policy

Reply:
Done! Thanks!